### DISCUSSION TITLE
error: Structures not same size

### DISCUSSION THREAD

**Discussion by ssunidhi** [charnley/rmsd#72] [Q&A] [answered]
Hey, amazing effort!

I have a quick question though.. can the code be modified to deal with different-sized proteins so that rmsd amongst different proteins can be found too?

**Comment by charnley** [comment, marked as answer]
Hi @ssunidhi, Unfortunately, the rotation matrix can only be defined for equal elements, same order & the same number of atoms. You'll have to define the atomic subsets of the two proteins you want to compare. 

If the two proteins are not at all the same, I will suggest looking at the RMSD of the alpha-carbons using QML atoms-reordering (see tests/ for example usage). It still requires the same number of carbon-alpha to work though.

Did that help answer your question?

**Comment by nbehrnd** [comment]
Hi @ssunidhi,

if your are on the way to compare the spatial arrangement of proteins
with each other, you might be better off with a program like PyMol.
Complementary to calculate_rmsd.py, Pymol allows you to compare
proteins with different atom connectivity, too and report a RMSD of
the fit determined.

Depending on the sequence similarity of the two proteins to be
compared with each other align, cealign, or super may be suitable
for you:

https://pymolwiki.org/index.php/Align
https://pymolwiki.org/index.php/Cealign
https://pymolwiki.org/index.php/Super

**Comment by krzys-r** [comment]
Actually it's quite often that we need to compare structure with different ligands, i.e. different atom count. 
It would be nice to have option to give additionally two lists of atoms that should be fitted (and the rest should be just rotate but not used in rmsd-computation).

**Reply by nbehrnd** [reply]
@krzys-r Do you have the ligand's structure models in a file format like .sdf, i.e. not only about the atomic coordinates, but including explicit bonds?

If so, your request *in part* relates to the identification of the «maximum common substructure» e.g. RDKit offers ([entry in the documentation](https://www.rdkit.org/docs/GettingStartedInPython.html#maximum-common-substructure)).  The group around Professor Andrea Volkamer set up a [talktorial](https://projects.volkamerlab.org/teachopencadd/talktorials/T006_compound_maximum_common_substructures.html) about this technique.

After the generation of such a (SMARTS encoded) «mask» , one possibly can then

+ remove all structure features in model A and B which do not correspond to this motif in common
+ export the truncated models in the .xyz format
+ eventually perform the Kabsch test with calculate_rmsd.py

**Reply by krzys-r** [reply]
When I wrote ligands I meant metalloorganic, which means SMARTS will probably not work. 
Anyway, the technique you describe will apparently allow to compute rmsd but for the publication it sometimes nice to plot both structure as overlapping (color coded or one dashed), to demonstrate differences. For that we need a final geometry with all atoms.

**Reply by charnley** [reply]
@krzys-r, can you open up an issue with example input files to describe your use case?

There is a development branch that looks at alpha carbons only but still requires the same number of alpha-carbons.

You can specify to include/exclude atoms on the command line. But you still need to choose which atoms to align.
